presto-parser/src/test/java/com/facebook/presto/sql/TestSqlTemplateFormatter.java

```diff
@@ -0,0 +1,158 @@
+    }
+
+    @Test
+    public void testSqlFormatterBooleanLiteral()
+    {
+        String statement = "SELECT * FROM table1 WHERE a = true AND b = false";
+        String templateStatement = "SELECT * FROM table1 WHERE a = ? AND b = ?";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterCharLiteral()
+    {
+        assertTemplate("SELECT * WHERE c = CHAR 'abc'", "SELECT * WHERE c = ?");
+    }
+
+    @Test
+    public void testSqlFormatterDecimalLiteral()
+    {
+        Statement statement = new Call(QualifiedName.of("foo"), ImmutableList.of(
+                new CallArgument("a", new DecimalLiteral("5")),
+                new CallArgument("b", new DecimalLiteral("6"))));
+        String templateStatement = "CALL foo(a => ?, b => ?)";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterDoubleLiteral()
+    {
+        Statement statement = new Call(QualifiedName.of("foo"), ImmutableList.of(
+                new CallArgument("a", new DoubleLiteral("5")),
+                new CallArgument("b", new DoubleLiteral("6"))));
+        String templateStatement = "CALL foo(a => ?, b => ?)";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testRewriteGenericLiteral()
+    {
+        Statement statement = new Call(QualifiedName.of("foo"), ImmutableList.of(
+                new CallArgument("a", new GenericLiteral("varchar", "test"))));
+        String templateStatement = "CALL foo(a => ?)";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterIntervalLiteral()
+    {
+        String statement = "SELECT * FROM table1 where Date1 < Date2 + INTERVAL '33' day to second";
+        String templateStatement = "SELECT * FROM table1 where Date1 < Date2 + ?";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterNullLiteral()
+    {
+        assertTemplate("SELECT * FROM table1 WHERE null", "SELECT * FROM table1 WHERE ?");
+    }
+
+    @Test
+    public void testSqlFormatterStringLiteral()
+    {
+        Statement statement = new Call(QualifiedName.of("foo"), ImmutableList.of(
+                new CallArgument("b", new StringLiteral("go"))));
+        String templateStatement = "CALL foo(b => ?)";
+        assertTemplate(statement, templateStatement);
+    }
+
+    @Test
+    public void testSqlFormatterTimeLiteral()
+    {
+        assertTemplate("SELECT * FROM table1 WHERE Time = '01:02:03.1234'", "SELECT * FROM table1 WHERE Time = ?");
+    }
+
+    @Test
+    public void testSqlFormatterTimestampLiteral()
+    {
+        assertTemplate("SELECT * FROM table1 WHERE DateTime > '2022-01-01 01:02:03.1234'", "SELECT * FROM table1 WHERE DateTime > ?");
+    }
+
+    @Test
+    public void testSqlFormatterMultipleLiteralTypesWithInnerJoinWithWhereWithGroupByWithHaving()
```

Comment:
This is a very long method name. Maybe let's call this "testTemplateComplexSql" or something like that and combine this with the next test.

Reply:
Updated accordingly